contracts/icp/context-config/src/types.rs

```diff
@@ -0,0 +1,322 @@
+use std::borrow::Cow;
+use std::collections::HashMap;
+
+use calimero_context_config::repr::{Repr, ReprBytes, ReprTransmute};
+use calimero_context_config::types::{
+    Application, ApplicationId, ApplicationMetadata, ApplicationSource, BlobId, Capability,
+    ContextId, SignerId,
+};
+use candid::CandidType;
+use ed25519_dalek::{Verifier, VerifyingKey};
+use serde::{Deserialize, Serialize};
+
+use crate::guard::Guard;
+
+#[derive(
+    CandidType, Serialize, Deserialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Copy, Hash,
+)]
+pub struct Identity(pub [u8; 32]);
+
+#[derive(
+    CandidType, Serialize, Deserialize, Clone, Debug, PartialEq, Eq, PartialOrd, Ord, Copy,
+)]
+pub struct ICSignerId(pub Identity);
+
+impl ICSignerId {
+    pub fn new(bytes: [u8; 32]) -> Self {
+        Self(Identity(bytes))
+    }
+}
+
+pub type ICContextIdentity = ICSignerId;
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug, Hash, Eq, PartialEq)]
+pub struct ICContextId(pub Identity);
+
+impl ICContextId {
+    pub fn new(bytes: [u8; 32]) -> Self {
+        Self(Identity(bytes))
+    }
+}
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct ICApplicationId(pub Identity);
+
+impl ICApplicationId {
+    pub fn new(bytes: [u8; 32]) -> Self {
+        Self(Identity(bytes))
+    }
+}
+
+#[derive(CandidType, Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub struct ICBlobId(pub Identity);
+
+impl ICBlobId {
+    pub fn new(bytes: [u8; 32]) -> Self {
+        Self(Identity(bytes))
+    }
+}
+
+// Update the From implementations
+impl From<SignerId> for ICSignerId {
+    fn from(value: SignerId) -> Self {
+        ICSignerId(Identity(value.as_bytes()))
+    }
+}
+
+impl From<Repr<SignerId>> for ICSignerId {
+    fn from(value: Repr<SignerId>) -> Self {
+        ICSignerId(Identity(value.as_bytes()))
+    }
+}
```

Comment:
no `U: From<Repr<T>>` needed if `U: From<T>`

```suggestion
```

you can remove the others as well